src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -619,6 +639,63 @@ class NetworkOPsImp final
     std::vector <TransactionStatus> mTransactions;
 
     StateAccounting accounting_ {};
+
+private:
+    struct Stats
+    {
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , disconnected_duration (collector->make_gauge("State_Accounting","Disconnected_duration"))
+            , connected_duration (collector->make_gauge("State_Accounting","Connected_duration"))
+            , syncing_duration(collector->make_gauge("State_Accounting", "Syncing_duration"))
+            , tracking_duration (collector->make_gauge("State_Accounting", "Tracking_duration"))
+            , full_duration (collector-> make_gauge("State_Accounting", "Full_duration"))
+            , disconnected_trasitions (collector->make_gauge("State_Accounting","Disconnected_trasitions"))
+            , connected_trasitions (collector->make_gauge("State_Accounting","Connected_trasitions"))
+            , syncing_trasitions(collector->make_gauge("State_Accounting", "Syncing_trasitions"))
+            , tracking_trasitions (collector->make_gauge("State_Accounting", "Tracking_trasitions"))
+            , full_trasitions (collector-> make_gauge("State_Accounting", "Full_trasitions"))
+            { }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge disconnected_duration;
+        beast::insight::Gauge connected_duration;
+        beast::insight::Gauge syncing_duration;
+        beast::insight::Gauge tracking_duration;
+        beast::insight::Gauge full_duration;
+
+        beast::insight::Gauge disconnected_trasitions;
+        beast::insight::Gauge connected_trasitions;
+        beast::insight::Gauge syncing_trasitions;
+        beast::insight::Gauge tracking_trasitions;
+        beast::insight::Gauge full_trasitions;
+    };
+
+    Stats m_stats;
+
+private:
+    void collect_metrics()
+    {   
+        auto counters = accounting_.getCounters();
+        auto start = accounting_.getStart();
+        auto mode = accounting_.getMode();
+
```

Comment:
You could use a small lambda to clean-up the casts. 
```
auto getCounters = [&counters_](auto mode) -> auto& {
    return counters_[static_cast<std::size_t>(mode)];
}
```
Then you can use it like...
```
getCounter(mode).dur += current;
m_stats.disconnected_duration.set(getCounter(OperatingMode::Disconnected).dur.count());
```

Not sure how much better that really is though, I'll let you decide. I think we try to keep the columns <= 80 chars.